qubesadmin/tests/mock_app.py

```diff
@@ -496,6 +581,37 @@ def __init__(self, qapp: QubesTest, dev_class: str,
         self.backend_vm = backend_vm
         self.attached = attached
 
+        self.update_calls()
+
+    def device_string(self):
+        if self.dev_class == 'block':
+            return f'{self.dev_id} mode=w device_node=/dev/{self.dev_id} ' \
+                   f'size=123456 description={self.description}\n'
+        elif self.dev_class == 'pci':
+            bus, rest = self.dev_id.split(":")
+            device, function = rest.split(".")
+            return f"{self.dev_id.replace(':', '_')} " \
+                   f"function='{function}' device='{device}' bus='{bus}' " \
+                   f"libvirt_name=pci_00_'{bus}{device}_{function}' " \
+                   f"product='{self.description}' " \
+                   f"description='{self.description}'\n"
```

Comment:
in the output in R4.3 I see less properties for PCI:
```
00_1f.3:0x8086:0x2930::p0c0500 device_id='0x8086:0x2930::p0c0500' port_id='00_1f.3' devclass='pci' backend_domain='dom0' product='82801I (ICH9 Family) SMBus Controller' vendor='Intel Corporation' interfaces='p0c0500'
```
BTW, the first field looks weird, should be just port_id, not port_id+device_id. Could be a bug... But other fields look ok in this response.